gtsam/linear/tests/testGaussianFactorGraph.cpp

```diff
@@ -73,37 +82,48 @@ TEST(GaussianFactorGraph, sparseJacobian) {
   //  5  6  7  0  0  8
   //  9 10  0 11 12 13
   //  0  0  0 14 15 16
-
-  // Expected - NOTE that we transpose this!
-  Matrix expectedT = (Matrix(16, 3) <<
-      1., 1., 2.,
-      1., 2., 4.,
-      1., 3., 6.,
-      2., 1.,10.,
-      2., 2.,12.,
-      2., 3.,14.,
-      1., 6., 8.,
-      2., 6.,16.,
-      3., 1.,18.,
-      3., 2.,20.,
-      3., 4.,22.,
-      3., 5.,24.,
-      4., 4.,28.,
-      4., 5.,30.,
-      3., 6.,26.,
-      4., 6.,32.).finished();
-
-  Matrix expected = expectedT.transpose();
-
   GaussianFactorGraph gfg;
   SharedDiagonal model = noiseModel::Isotropic::Sigma(2, 0.5);
-  gfg.add(0, (Matrix(2, 3) << 1., 2., 3., 5., 6., 7.).finished(), Vector2(4., 8.), model);
-  gfg.add(0, (Matrix(2, 3) << 9., 10., 0., 0., 0., 0.).finished(), 1,
-          (Matrix(2, 2) << 11., 12., 14., 15.).finished(), Vector2(13., 16.), model);
-
-  Matrix actual = gfg.sparseJacobian_();
-
-  EXPECT(assert_equal(expected, actual));
+  const Key x123 = 0, x45 = 1;
+  gfg.add(x123, (Matrix(2, 3) << 1, 2, 3, 5, 6, 7).finished(),
+          Vector2(4, 8), model);
+  gfg.add(x123, (Matrix(2, 3) << 9, 10, 0, 0, 0, 0).finished(),
+          x45,  (Matrix(2, 2) << 11, 12, 14, 15.).finished(),
+          Vector2(13, 16), model);
+
+  // Check version for MATLAB - NOTE that we transpose this!
```

Comment:
Diff is weird. Can you move back up to minimize diff?

Reply:
done